### PR TITLE
Feature obligated exceeds potential

### PR DESCRIPTION
**Pull request by ebdabbs**
### \*Merge after #1042 \*

**High level description:**

Implements the bar graph for the "Extreme Overspending" case in the IDV aggregate Award Amounts visualization. 

**Technical details:**
- Added `extremeOverspending` and `extremeOverspendingFormatted` getters to the model. 
- Perform a check for the situation where obligated amounts exceed combined potential award amounts and implements the corresponding bar graph.
- Award id for an example of extreme overspending: `69188958` 

**JIRA Ticket:**
[DEV-2358](https://federal-spending-transparency.atlassian.net/browse/DEV-2358), subtask for step 4 of [DEV-2224](https://federal-spending-transparency.atlassian.net/browse/DEV-2224)

**Mockup:**
https://bahdigital.invisionapp.com/share/69IA8EPGPCM#/296009662_combinedAwardsSection--overspend2

The following are ALL required for the PR to be merged:
- [x] #1042 merged, change base to `dev`
- [x] Add tooltip
- [x] Code review
- [x] Design review
- [x] Verified cross-browser compatibility

### CODE REVIEW

File: src/js/components/awardv2/visualizations/amounts/charts/ExceedsPotentialChart.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+};
+
+export default class ExceedsPotentialChart extends React.Component {
+    render() {
+        const obligation = this.props.awardAmounts._obligation;
+        const current = this.props.awardAmounts._combinedCurrentAwardAmounts;
+        const potential = this.props.awardAmounts._combinedPotentialAwardAmounts;
+
+        const overspendingBarStyle = {
+            width: generatePercentage((obligation - potential) / obligation)
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/visualizations/amounts/charts/ExceedsPotentialChart.jsx]: looks like we need a closing paren here.

File: src/js/models/v2/awardsV2/BaseAwardAmounts.js [hunk 2]
```diff
@@ -60,6 +60,16 @@ const BaseAwardAmounts = {
             return `${MoneyFormatter.formatMoneyWithPrecision((this._obligation - this._combinedCurrentAwardAmounts) / units.unit, 1)} ${units.unitLabel}`;
         }
         return MoneyFormatter.formatMoney(this._obligation - this._combinedCurrentAwardAmounts);
+    },
+    get extremeOverspending() {
+        return MoneyFormatter.formatMoneyWithPrecision(this._obligation - this._combinedPotentialAwardAmounts, 2);
+    },
+    get extremeOverspendingFormatted() {
```
Comment (MaxwellKendall) [inline src/js/models/v2/awardsV2/BaseAwardAmounts.js]: Where are we using these?

File: src/js/models/v2/awardsV2/BaseAwardAmounts.js [hunk 3]
```diff
@@ -60,6 +60,16 @@ const BaseAwardAmounts = {
             return `${MoneyFormatter.formatMoneyWithPrecision((this._obligation - this._combinedCurrentAwardAmounts) / units.unit, 1)} ${units.unitLabel}`;
         }
         return MoneyFormatter.formatMoney(this._obligation - this._combinedCurrentAwardAmounts);
+    },
+    get extremeOverspending() {
```
Comment (MaxwellKendall) [inline src/js/models/v2/awardsV2/BaseAwardAmounts.js]: Where are we using these?

File: src/js/models/v2/awardsV2/BaseAwardAmounts.js (hunk 2 above)
Comment (ebdabbs) [reply src/js/models/v2/awardsV2/BaseAwardAmounts.js]: `extremeOverspendingFormatted` is used [here](https://github.com/fedspendingtransparency/usaspending-website/blob/feature-obligated-exceeds-potential/src/js/components/awardv2/visualizations/amounts/charts/ExceedsPotentialChart.jsx#L63).

File: src/js/models/v2/awardsV2/BaseAwardAmounts.js (hunk 3 above)
Comment (ebdabbs) [reply src/js/models/v2/awardsV2/BaseAwardAmounts.js]: `extremeOverspending` is used [here](https://github.com/fedspendingtransparency/usaspending-website/blob/feature-obligated-exceeds-potential/src/js/components/awardv2/visualizations/amounts/AggregatedAwardAmounts.jsx#L59) (in the table).

File: src/js/components/awardv2/visualizations/amounts/charts/ExceedsPotentialChart.jsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { generatePercentage } from 'helpers/aggregatedAmountsHelper';
+
+import InfoTooltip from 'components/awardv2/idv/InfoTooltip';
+import { awardAmountsExtremeOverspendingInfo } from 'components/awardv2/idv/InfoTooltipContent';
+
+const propTypes = {
+    awardAmounts: PropTypes.object
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/visualizations/amounts/charts/ExceedsPotentialChart.jsx]: Can we provide a new propType definition for this?

**Comment by MaxwellKendall** [comment]
@ebdabbs ,

Merged #1042 and changed the base branch, looks like there are some conflicts here now, or something? Only feedback I have is to maybe provide a propType definition for `awardAmounts`.

**Pull request merged**